### DISCUSSION TITLE
Time dep

### DISCUSSION THREAD

**Discussion by ragusa** [chi-tech/chi-tech#278] [Updates to `development`]
Trying to see if a discussion is a good means to discuss the time-dep new features, post PR.

**Comment by zachhardy** [comment]
Anything works! Anything in particular you waned to discuss?

**Comment by ragusa** [comment]
Yes but since I came too late, after the merge, I absolutely do not know how to start a discussion that also comments the code. So I’ll go in this email:

lbts_01_initialize.cc

  *   Line 16: post a comment to explain why we divide by keff.
     *   Usually, it is to have the initial fission source normalized to that a null-transient will run where the total flux remains constant.
     *   However, talking with Jan, he told me that it was that the fission rate was computed with phi_old and not updated upon convergence of the Keigenvalue solver, hence there might be a slight difference in flux and thus in the normalization of the FR.
     *   If Jan’s point is valid, and it may very well be, then how do we ensure running null transients?
     *   Regardless, please add a short comment to the code
  *   Lines 30-32. People build (implicitly) mental construct to understand things. Here, on the left hand side, all names end with “_prev_local”. However, on the right hand side, what you are coying into the _prev_local variables is either _old_local or _new_local and this is confusing me.
  *   Line 58: comment a bit more on the active_set_src … and why the “using namespace std::placeholders;” is necessary

lbts_02_execute.cc

  *   At some point, it may be good to also have a PreSetCallBack. MOOSE has both Pre and PostSteps. Just FYI
  *   Line 40:
     *   it took me a few seconds to understand _prev is actually the previous time step values (or ‘old’ time step values, but we should not say ‘old’)
     *   ‘old’ for you is reserved to the values at the current time, but from the older/previous iteration.
     *   If I am right, it may be worth to spell this out somewhere.
        *   Previous = at time T- delt T
        *   Old = at time T, but old iterate
        *   New = at time T, current or new iterate
  *   Line 55: couldn’t this line be removed? I do not see where it is used.
  *   Lines 126-127: now, the new values get copied into the prev values (which was not done like that in the init call, see above)

Before you decide to add TR-BDF2 and other implicit time marching, let’s discuss. There is a way to be very generic with Butcher tableaux and most methods and fir that general framework.

**Reply by zachhardy** [reply]
I agree that `_prev` and `_old` are a bit confusing. We did this because `_old` was already used for the previous iterate in the steady-state solver. My preference would be to either flip the definitions `_old` to `_prev` and vice versa, or to change `_old` to `_ell` or `_k` and `_prev` to `_old`. I can perhaps go through and make some of the simple changes in the coming days.

**Comment by ragusa** [comment]
Ok. Add a comment in the code to state the terminology once you settle on what is the most natural.
